tests/jballerina-unit-test/src/test/resources/test-src/types/table/table-value-negative.bal

```diff
@@ -47,3 +45,55 @@ function testVariableNameFieldAsKeyField() {
         {idNum: 2, name: "Amy"}
     ];
 }
+
+function testVariableNameFieldAsKeyField2() {
+    table<record {readonly int idNum = 2; string name;}> key (idNum) _ = table [
+        {idNum, name: "Jo"},
+        {idNum, name: "Chiran"},
+        {idNum: 2, name: "Amy"}
+    ];
+}
+
+function testVariableNameFieldAsKeyField3() {
+    table<record {readonly int idNum = 2; readonly string name;}> key (idNum, name) _ = table [
+        {idNum, name: "Jo"},
+        {idNum, name: "Jo"},
+        {idNum: 2, name: "Amy"}
+    ];
+}
+
+function testVariableNameFieldAsKeyField4() {
+    table<record {readonly int idNum = 2; readonly string name = "A";}> key (idNum, name) _ = table [
+        {idNum, name: "Jo"},
+        {idNum, name: "Jo"},
+        {idNum: 2, name: "Amy"}
+    ];
+}
+
+type Foo2 record {
+    readonly map<string> m = {};
+    int age;
+};
+
+type GlobalTable3 table<Foo2> key(m);
+
+function testTableConstructExprWithDuplicateKeys2() {
+    GlobalTable3 _ = table [
+      {m: {"AAA": "DDDD"}, age: 31},
+      {m: {"AAA": "DDDD"}, age: 34}
+    ];
```

Comment:
Formatting seems off. Check other places too.

Reply:
According to https://github.com/ballerina-platform/ballerina-lang/pull/37491#discussion_r1013636630
This has correct formatting ryt?.

Reply:
Indentation seems off.

Reply:
Updated